addons/sys_prc117f/farris_menus/PGM.sqf

```diff
@@ -399,8 +402,15 @@ GVAR(PGM_NORM_LOS) = ["PGM_NORM_LOS", "PGM_NORM_LOS", "",
             ],
             [
                 {
-                    private _value = nil;
-                    _value = GET_RADIO_VALUE("name");
+                    private _value = GET_RADIO_VALUE("name");
+                    private _valueLen = count _value;
+                    if (_valueLen < CHANNELNAME_MAX_LEN) then {
+                        TRACE_1("Resizing channel name", _valueLen);
+                        private _whitespace = [];
+                        _whitespace resize CHANNELNAME_MAX_LEN - _valueLen;
+                        _whitespace = _whitespace apply {SPACE_CHAR};
+                        _value = [_value, toString _whitespace] joinString "";
+                    };
```

Comment:
This approach seems pretty long winded, couldn't you just do `_value = _value select [0, 20]` to substring the string? This would then also remove the need to trim?

Reply:
No, that wouldn't work. If you use a length greater than the length of the string the result just gets truncated to the length of the string. The whole idea is to extend the string so that when a character is inserted there actually is a spot to store it. If not there is either no array element there or the string will have nils in between.

For example if a channel is named "NET01" and I want to name it "NET01 A" you get a zero divisor because there is no more array:
[ N, E, T, 0, 1 ]   A
                         ^

Reply:
Good point! I wouldn't be surprised if something like this was much faster though (just appending 20 spaces and substring)
```
_value = (_value + "                    ") select [0,20]; // Append 20 spaces to pad end with spaces and substring to length 20
```

Reply:
0.0033 ms over 0.0012 ms
It's a little magic number-y but it works. Changing the maximum channel length would then require hard coded changes in three places. Is that worth the 0.0021 ms the first time the menu is opened?

Reply:
Max characters for channel names should be 12 for both 152 and 117F. And shall defer decision on which approach for padding the string to use to @jonpas

Reply:
> Changing the maximum channel length would then require hard coded changes in three places.

Not a problem, `#define` it in `script_component.hpp` of each of the 2 radios and you are done. All radio menus are designed as per real life, their parameters don't change dynamically, so there is no reason to waste performance or readability.

Reply:
You want the actual padding string defined too?

Reply:
If the channel name max length is going into a define, then it would probably make sense to define the padded string with it so that both declarations in the same place.

Otherwise the alternative would be to have a longer padded string that could handle a longer max length but that a negative of that is that it is still arbitrary. However the max length is probably never going to subsequently change once it is made 12 (which is why personally I wouldn't macro it). I am willing to accept macros though.

Reply:
All magic numbers should be macros.
Macros aren't there only for easy changing, but also for readability.

Reply:
Updated